ddtrace/tracer/option.go

```diff
@@ -186,16 +187,23 @@ func WithPropagator(p Propagator) StartOption {
 	}
 }
 
-// WithServiceName sets the default service name to be used with the tracer. It is
-// deprecated in favour of WithService and will be removed in the next major version.
+// WithServiceName sets the default service name to be used with the tracer. This will
+// not affect the service name used by any integrations. It is deprecated in favour of
+// WithService and will be removed in the next major version.
```

Comment:
I would like to suggest:
```go
// WithServiceName is deprecated. Please use WithService. 
// If you are using an older version and you are upgrading from WithServiceName
// to WithService, please note that WithService will determine server and framework
// integrations to share the service name. This provides a better experience of the
// product.
```
We should make sure to explain the same thing in the release notes, along with clear explanations about why the new form is better (@brettlangdon get ready to help with this 🙂)

Reply:
Made a minor modification, but 👍